### PR TITLE
Add availability zone to be used by controller plane

### LINKED ISSUE #525: AZ usage of v1alpha3
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

**What did you expect to happen:**
my cloud actually have more than 1 az, and I noticed that current behavior is to create vm on random az , I checked cluster-api code and seems their az is try to find in a set of domains then pick fewest one

but openstack az is not such concept as far as I can tell... 
are we sure the cluster-api failuredomain concept is same to az in openstack? 
as cluster api has scale up and down ,I am really not convieneced it's suitable for openstack

and , where can I set failuredomain on control pane? I knew we can set on worker node but I am not that far 

```

func (r *KubeadmControlPlaneReconciler) failureDomainForScaleDown(cluster *clusterv1.Cluster, machines internal.FilterableMachineCollection) *string {
        // See if there are any Machines that are not in currently defined failure domains first.
        notInFailureDomains := machines.Filter(
                machinefilters.Not(machinefilters.InFailureDomains(cluster.Status.FailureDomains.FilterControlPlane().GetIDs()...)),
        )
        if len(notInFailureDomains) > 0 {
                // return the failure domain for the oldest Machine not in the current list of failure domains
                // this could be either nil (no failure domain defined) or a failure domain that is no longer defined
                // in the cluster status.
                return notInFailureDomains.Oldest().Spec.FailureDomain
        }

        // Otherwise pick the currently known failure domain with the most Machines
        return internal.PickMost(cluster.Status.FailureDomains.FilterControlPlane(), machines)
}

func (r *KubeadmControlPlaneReconciler) failureDomainForScaleUp(cluster *clusterv1.Cluster, machines internal.FilterableMachineCollection) *string {
        // Don't do anything if there are no failure domains defined on the cluster.
        if len(cluster.Status.FailureDomains.FilterControlPlane()) == 0 {
                return nil
        }
        return internal.PickFewest(cluster.Status.FailureDomains.FilterControlPlane(), machines)
}
```

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #525 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: api/v1alpha3/openstackcluster_types.go [hunk 1]
```diff
@@ -115,6 +115,9 @@ type OpenStackClusterSpec struct {
 	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
 	// +optional
 	ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+
+	// ControlPlaneAvailabilityZones is the az to deploy controller to
```
Comment (sbueringer) [inline api/v1alpha3/openstackcluster_types.go:119]: Can you replace controller with control plane nodes?

**Comment by sbueringer** [comment]
recheck

**Comment by sbueringer** [comment]
recheck

File: api/v1alpha3/openstackcluster_types.go (hunk 1 above)
Comment (sbueringer) [reply api/v1alpha3/openstackcluster_types.go]: Regen is missing

**Comment by sbueringer** [comment]
@jichenjc you have to execute: `make generate-manifests`

**Comment by jichenjc** [comment]
yeah , I am still stuggling to restore my env
I want to give a real try before remove the hold flag and also update with the command you provided :)

**Pull request merged**